platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py

```diff
@@ -0,0 +1,351 @@
+        THERMAL_DEV_CATEGORY_AMBIENT:(0,
+            [
+                THERMAL_DEV_ASIC_AMBIENT,
+                THERMAL_DEV_COMEX_AMBIENT,
+                THERMAL_DEV_PORT_AMBIENT,
+                THERMAL_DEV_FAN_AMBIENT
+            ]
+        )
+    },
+]
+
+def initialize_thermals(sku, thermal_list, psu_list):
+    tp_index = hwsku_dict_thermal[sku]
+    thermal_profile = thermal_profile_list[tp_index]
+    for category in thermal_device_categories_all:
+        if category == THERMAL_DEV_CATEGORY_AMBIENT:
+            count, ambient_list = thermal_profile[category]
+            for ambient in ambient_list:
+                thermal = Thermal(category, ambient, True)
+                thermal_list.append(thermal)
+        else:
+            start, count = 0, 0
+            if category in thermal_profile:
+                start, count = thermal_profile[category]
+                if count == 0:
+                    continue
+            if count == 1:
+                thermal = Thermal(category, 0, False)
+                thermal_list.append(thermal)
+            else:
+                if category == THERMAL_DEV_CATEGORY_PSU:
+                    for index in range(count):
+                        thermal = Thermal(category, start + index, True, psu_list[index])
+                        thermal_list.append(thermal)
+                else:
+                    for index in range(count):
+                        thermal = Thermal(category, start + index, True)
+                        thermal_list.append(thermal)
+
+class Thermal(ThermalBase):
+    def __init__(self, category, index, has_index, dependency = None):
+        """
+        index should be a string for category ambient and int for other categories
+        """
+        if category == THERMAL_DEV_CATEGORY_AMBIENT:
+            self.name = thermal_ambient_name[index]
+            self.index = index
+        elif has_index:
+            self.name = thermal_name[category].format(index)
+            self.index = index
+        else:
+            self.name = thermal_name[category]
+            self.index = 0
+
+        self.category = category
+        self.temperature = self._get_file_from_api(THERMAL_API_GET_TEMPERATURE)
+        self.high_threshold = self._get_file_from_api(THERMAL_API_GET_HIGH_THRESHOLD)
+        self.dependency = dependency
+
+    def get_name(self):
+        """
+        Retrieves the name of the device
+
+        Returns:
+            string: The name of the device
+        """
+        return self.name
+
+    def _read_generic_file(self, filename, len):
+        """
+        Read a generic file, returns the contents of the file
+        """
+        result = None
+        try:
+            with open(filename, 'r') as fileobj:
+                result = fileobj.read()
+        except:
+            if self.dependency is None or self.dependency.get_powergood_status():
+                logger.log_warning("Fail to read file {}, maybe it doesn't exist".format(filename))
```

Comment:
It is better to have the accurate logs for the case that power is off for troubleshooting purpose.

Reply:
fixed. 
in case of power off, another piece of information "power off" is logged. 
only when the file should exist but doesn't is this information logged.